src/features/settings/screens/camera.screen.js

```diff
@@ -1,3 +1,35 @@
-import React from "react";
+import React, { useRef, useState, useEffect } from "react";
+import { Camera } from "expo-camera";
+import styled from "styled-components/native";
+import { View } from "react-native";
+import { Text } from "../../../components/typography/text.component";
 
-export const CameraScreen = () => null;
+const ProfileCamera = styled(Camera)`
+  width: 100%;
+  height: 100%;
+`;
+
+export const CameraScreen = () => {
+  const [hasPermission, setHasPermission] = useState(null);
+  const cameraRef = useRef();
+
+  useEffect(() => {
+    (async () => {
+      const { status } = await Camera.requestPermissionsAsync();
+      setHasPermission(status === "granted");
+    })();
+  }, []);
+
+  if (hasPermission === null) {
+    return <View />;
+  }
+  if (hasPermission === false) {
+    return <Text>No access to camera</Text>;
+  }
+  return (
+    <ProfileCamera
+      ref={(camera) => (cameraRef.current = camera)}
+      type={Camera.Constants.Type.front}
```

Comment:
Use this snippet instead, new expo Camera API

[https://docs.expo.dev/versions/v45.0.0/sdk/camera/](https://docs.expo.dev/versions/v45.0.0/sdk/camera/)
```
import { Camera, CameraType } from 'expo-camera';

return (
      <ProfileCamera
         ref={camera => (cameraRef.current = camera)}
         type={CameraType.front}
      ></ProfileCamera>
   );
```

Reply:
If you want a fully Typed version of the app with Typescript go to 

[https://github.com/ed3899/meals-to-go](https://github.com/ed3899/meals-to-go)

Look through the branches and find the section and commit you're looking for. Some of them are not sequential (adding typings, slightly changing component because of typescript warning, es lint for typescript-react-native, etc). But for the most part they resemble the original project.